envi/archs/arm/disasm.py

```diff
@@ -2889,119 +2900,445 @@ def _do_adv_simd_ldst_32(val, va, u):
+                            )
+
 
             elif optype in (0b0000, 0b0001):
                 # vst4
                 mnem = 'vst4'
+                inc = 1 + (optype&1)
+                if reg_index:
+                    pubwl &= ~0x10
+                    opers = (
+                            ArmExtRegListOper(dd, count, 1, inc=inc),    # size in this context means use "D#" registers
+                            ArmRegOffsetOper(rn, rm, va, pubwl=pubwl, psize=size, tsize=size, align=align),
+                            )
+                else:
+                    if rm == 0xd:
+                        pubwl |= 0x2
+                    opers = (
+                            ArmExtRegListOper(dd, count, 1, inc=inc),
+                            ArmImmOffsetOper(rn, 0, va, pubwl=pubwl, psize=size, tsize=size, align=align),
+                            )
+
+
 
         else:
             # single elements
+            optype = (val >> 8) & 0x3
+            szidx = (val>>10) & 3    # different location for a=0 and a=1
+            sflag, ebytes =  ((IFS_8,1), (IFS_16,2), (IFS_32,4), (None, None))[szidx]
+
             index_align = (val >> 4) & 0xf
-            size = (val >> 10) & 3
+            count = (1,2,3,4,) [ optype ]
+            index = index_align >> (1+szidx)
 
-            if optype in (0b0000, 0b0100, 0b1000):
-                # vst1
+            # set increment size
+            if (index_align >> szidx) == 0:
+                inc = 1
+            else:
+                inc = (1,2,2)[szidx]
+
+            if optype == 0b00:
+                # vst1  - single 1-element structure from One lane
                 mnem = 'vst1'
-                opers = (
-                        ArmExtRegListOper(dd, count, 1),
-                        ArmImmOffsetOper(rn, 0, va, pubwl=pubwl, psize=size, tsize=size),
-                        )
-            elif optype in (0b0001, 0b0101, 0b1001):
-                # vst2
+                if index_align & 1 == 0:
+                    align = 1
+                else:
+                    align = ebytes
+
+                if reg_index:
+                    pubwl &= ~0x10
+                    opers = (
+                            ArmExtRegListOper(dd, count, 1, index=index),
+                            ArmRegOffsetOper(rn, rm, va, pubwl=pubwl, psize=ebytes, tsize=ebytes, align=align),
+                            )
+                else:
+                    if rm == 0xd:
+                        pubwl |= 0x2
+                    opers = (
+                            ArmExtRegListOper(dd, count, 1, index=index),
+                            ArmImmOffsetOper(rn, 0, va, pubwl=pubwl, psize=ebytes, tsize=ebytes, align=align),
+                            )
+
+            elif optype == 0b01:
+                # vst2  - single 2-element structure from One lane
+                #TODO:  ALIGN  and AlignedRegisterOper
```

Comment:
Is this a super pressing TODO or nah?

Reply:
lol, i got out the manual, looked up what i meant by that, implemented it... only to discover that i already had it done. <facepalm>